src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/INodeFilter.cs

```diff
@@ -0,0 +1,72 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Nodes
+{
+    /// <summary>
+    /// Has information for filtering a SMO object by properties 
+    /// </summary>
+    public interface INodeFilter
+    {
+        /// <summary>
+        /// Returns true if the filter can be apply to the given type and Server type
+        /// </summary>
+        /// <param name="type">Type of the querier</param>
+        /// <param name="validForFlag">Server Type</param>
+        /// <returns></returns>
+        bool CanApplyFilter(Type type, ValidForFlag validForFlag);
+
+        /// <summary>
+        /// Creates a string from the filter property and values to be used in the Urn to query the SQL objects
+        /// Example of the output:[@ IsSystemObject = 0]
+        /// </summary>
+        /// <returns></returns>
+        string ToPropertyFilterString(Type type, ValidForFlag validForFlag);
+
+        public static string GetPropertyFilter(IEnumerable<INodeFilter> filters, Type type, ValidForFlag validForFlag)
+        {
+            string filter = "";
+            var list = filters.ToList();
+            for (int i = 0; i < list.Count; i++)
+            {
+                var value = list[i];
+
+                var propertyFilterString = value.ToPropertyFilterString(type, validForFlag);
+                if (propertyFilterString != string.Empty)
+                {
+                    string andPrefix = filter == string.Empty ? string.Empty : "and";
+                    filter = $"{filter} {andPrefix} {propertyFilterString}";
```

Comment:
If I read this correctly, with this method, if we have 4 filter strings, the output will be something like this:
`[{{{{filter1} and {filter2}} and {filter3}} and {filter4}}]`

will `[{filter1} and {filter2} and {filter3} and {filter4}]` work this same way? if so, i'd recommend to use the latter one. easier to read and easier to use StringBuilder.